util/metadata.go

```diff
@@ -0,0 +1,43 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"	
+	"io/ioutil"
+	"strings"
+)
+
+func CheckError(err error) {
```

Comment:
Nit: It's idiomatic to repeat the `if err != nil` check numerous times in Go instead of wrapping it in a function. Despite repetitive, it's natural and somewhat reassuring for Go programmers. If you absolutely want to have a function for error checking, you probably don't want to export it.

Reply:
noted!